src/ngx_http_lua_bodyfilterby.c

```diff
@@ -136,7 +141,8 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
-    if (rc == NGX_ERROR || rc >= NGX_HTTP_SPECIAL_RESPONSE) {
+    if (ctx->body_filter_aborted || rc == NGX_ERROR
+        || rc >= NGX_HTTP_SPECIAL_RESPONSE) {
```

Comment:
style: the following style can be better.
```suggestion
        if (ctx->body_filter_aborted
            || rc == NGX_ERROR
            || rc >= NGX_HTTP_SPECIAL_RESPONSE)
        {
```